src/test/scala-2/chiselTests/DomainSpec.scala

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiselTests
+
+import chisel3._
+import chisel3.domain.{Domain, Field}
+import chisel3.domains.ClockDomain
+import chisel3.experimental.ExtModule
+import chisel3.testing.FileCheck
+import circt.stage.ChiselStage
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+class DomainSpec extends AnyFlatSpec with Matchers with FileCheck {
+
+  behavior of "Domains"
+
+  they should "emit FIRRTL for internal and user-defined domains" in {
+
+    object PowerDomain extends Domain {
+
+      override def fields = Seq(
+        ("name", Field.String),
+        ("voltage", Field.Integer),
+        ("alwaysOn", Field.Boolean)
+      )
+
+    }
+
+    class Foo extends RawModule {
+      val A = IO(Input(domain.Type(ClockDomain)))
+      val B = IO(Input(domain.Type(PowerDomain)))
```

Comment:
This feels a bit inconsistent with the rest of Chisel, what about:
```suggestion
      val A = IO(Input(ClockDomain()))
      val B = IO(Input(PowerDomain()))
```
or perhaps
```suggestion
      val A = IO(Input(ClockDomain.Type()))
      val B = IO(Input(PowerDomain.Type()))
```

Reply:
This is a nice idea. This did feel somewhat awkward... I switched to your suggestion in: c8eb82305

WDYT about `.Type()` vs `.type()`?